lib/shackles.rb

```diff
@@ -2,8 +2,11 @@
 
 module Shackles
   class << self
+    attr_accessor :primary_environment_name
+    Shackles.primary_environment_name = :primary
```

Comment:
I had to call `Shackles.primary_environment_name` instead of `self.primary_environment_name` to avoid this error:

```
NoMethodError:
  undefined method `primary_environment_name=' for #<Class:Shackles>
```

which I'm not fully grokking. When I run this

```ruby
module Shackles
  puts self
  class << self
    puts self
    puts Shackles
...
```

I get this:

```
Shackles
#<Class:Shackles>
Shackles
```

My Ruby is pretty rusty... why the `#<Class:Shackles>` for `self` in the `class << self` context? And why ins't that referring to the same object that `attr_accessor` is adding methods to?

Reply:
ah yes, I should have caught that. inside the `class << self` block, `self` refers to the singleton class. which is what you address to add the instance variable to the class. but then you need to call the class method (not the singleton class method) to assign it.